kafka-connect/src/main/java/com/hazelcast/jet/contrib/connect/KafkaConnectSources.java

```diff
@@ -140,9 +193,12 @@ void fillBuffer(TimestampedSourceBuffer<SourceRecord> buf) {
                 }
 
                 for (SourceRecord record : records) {
-                    long ts = record.timestamp() == null ? 0 : record.timestamp();
-                    buf.add(record, ts);
-                    partitionsToOffset.put(record.sourcePartition(), record.sourceOffset());
+                    T projection = dataProjectionFn.apply(projectionContext, record);
+                    if (projection != null) {
+                        long ts = timestampProjectionFn.apply(projectionContext, projection);
```

Comment:
so we apply the timestamp function to the projected value, why not to the record itself?

Reply:
There is a timestamp in the record, which is no good, doesn't contain any value. So we get the timestamp from the parsed form of the JSON contained in the record, which means we would be doing the parsing twice (one for the timestamp, once for data mapping), that's what I wanted to avoid. However, what I have not considered until now is that maybe the record timestamp missing is a bug in our Kafka Connect source and maybe I can fix it there, making this projection unnecessary. Will look into that.

Reply:
No, unfortunately I don't have a better solution. Looked through the Debezium code, but seems like they just didn't bother with setting the Kafka ``SourceRecord``'s timestamp... You have to dig it out from the change event's value, which means parsing. So my changes still seem the most efficient solution.